common/workerpool.c

```diff
@@ -115,7 +116,8 @@ workerpool_t *workerpool_create(int nthreads)
             int res = pthread_create(&wp->threads[i], NULL, worker_thread, wp);
             if (res != 0) {
                 perror("pthread_create");
-                exit(-1);
+                // errno already set to EAGAIN by pthread_create() failure
+                return NULL;
```

Comment:
This is still printing on the terminal via [`perror`](https://en.cppreference.com/w/c/io/perror). Do we want the library to print to stderr?